source/shared/cpp/ObjectModel/BaseElement.h

```diff
@@ -84,23 +86,18 @@ namespace AdaptiveSharedNamespace
         // Fallback and Requires support
         FallbackType GetFallbackType() const { return m_fallbackType; }
         std::shared_ptr<BaseElement> GetFallbackContent() const { return m_fallbackContent; }
-        bool CanFallbackToAncestor() const { return m_canFallbackToAncestor; }
         void SetFallbackType(FallbackType type) { m_fallbackType = type; }
-        void SetFallbackContent(std::shared_ptr<BaseElement> element)
-        {
-            m_fallbackContent = element;
-        }
+        void SetFallbackContent(std::shared_ptr<BaseElement> element) { m_fallbackContent = element; }
 
-        bool MeetsRequirements(const std::unordered_map<std::string, std::string>& hostProvides) const;
+        bool MeetsRequirements(const AdaptiveSharedNamespace::FeatureRegistration& hostProvides) const;
+        std::shared_ptr<std::unordered_map<std::string, AdaptiveSharedNamespace::SemanticVersion>> GetRequirements() const;
 
         // Misc.
         virtual void GetResourceInformation(std::vector<RemoteResourceInformation>& resourceUris);
 
     protected:
         virtual void PopulateKnownPropertiesSet();
         void SetTypeString(const std::string& type) { m_typeString = type; }
```

Comment:
>std::shared_ptr<std::unordered_map<std::string, AdaptiveSharedNamespace::SemanticVersion>> m_requires; [](start = 8, length = 102)

why is this a pointer to an unordered_map instead of the unordered_map by itself?

Reply:
it was originally an `unordered_map`, but was turned into a `shared_ptr<unordered_map>` on @RebeccaAnne 's suggestion

---
In reply to: [276843946](https://github.com/Microsoft/AdaptiveCards/pull/2684#discussion_r276843946) [](ancestors = 276843946)